UndertaleModTool/Editors/UndertaleEmbeddedTextureEditor.xaml.cs

```diff
@@ -154,29 +174,41 @@ private void Import_Click(object sender, RoutedEventArgs e)
             {
                 try
                 {
-                    Bitmap bmp;
-                    using (var ms = new MemoryStream(TextureWorker.ReadTextureBlob(dlg.FileName)))
+                    GMImage image;
+                    if (System.IO.Path.GetExtension(dlg.FileName).Equals(".png", StringComparison.InvariantCultureIgnoreCase))
                     {
-                        bmp = new Bitmap(ms);
+                        // Import PNG data verbatim, without attempting to modify it
+                        image = GMImage.FromPng(File.ReadAllBytes(dlg.FileName), true).ConvertToFormat(target.TextureData.Image.Format);
```

Comment:
I would split this line like this:
```suggestion
image = GMImage.FromPng(File.ReadAllBytes(dlg.FileName), true)
               .ConvertToFormat(target.TextureData.Image.Format);
```

![image](https://github.com/user-attachments/assets/c40802e7-1c2e-4c08-ab48-84ce179ce3c8)